src/array.rs

```diff
@@ -135,93 +137,64 @@ unsafe impl<T: Element, D: Dimension> PyTypeInfo for PyArray<T, D> {
     }
 }
 
-pyobject_native_type_named!(PyArray<T, D>; T; D);
+pyobject_native_type_base!(PyArray<T, D>; T; D);
```

Comment:
@davidhewitt I am somewhat unhappy with open coding the boiler plate contained in `pyobject_native_type_named!`, but I saw no other away of achieving `impl Deref<Target=PyAnyArray> for PyArray<T, D>` as that impl conflicts with `impl Deref<Target=PyAny> for PyArray<T, D>` which would be part of `pyobject_native_type_named!`. And the alternative of duplicating he various method why apply to both `PyAnyArray` and `PyArray<T, D>` seemed even less palatable.